frontend/app/categories/page.tsx

```diff
@@ -37,22 +39,30 @@ interface Category {
   post_count: number;
   creator: User;
 }
+
 export default function ManageCategories() {
   const [categories, setCategories] = useState<Category[]>([]);
   const [open, setOpen] = useState(false);
   const [categoryName, setCategoryName] = useState("");
   const [editingCategory, setEditingCategory] = useState<Category | null>(null);
+  const [page, setPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const limit = 10; // Number of items per page
 
   useEffect(() => {
     fetchCategories();
-  }, []);
+  }, [page]);
 
   const fetchCategories = async () => {
     try {
-      const result = await getCategories();
-      setCategories(result.categories);
+      const result = await getCategories(page, limit);
+      setCategories(result.categories || []);
+      const total = result.total || 0;
+      setTotalPages(Math.max(1, Math.ceil(total / limit)));
```

Comment:
There is a result.pagination object that has this already calculated, let's reuse that please